lib/datadog/tracing/contrib/rack/middlewares.rb

```diff
@@ -93,13 +93,38 @@ def call(env)
             request_span.resource = nil
             request_trace = Tracing.active_trace
             env[Ext::RACK_ENV_REQUEST_SPAN] = request_span
+            Thread.current[:datadog_http_routing] = []
 
             # Copy the original env, before the rest of the stack executes.
             # Values may change; we want values before that happens.
             original_env = env.dup
 
             # call the rest of the stack
             status, headers, response = @app.call(env)
+
+            # TODO: might be wrong if say rails routed to sinatra but sinatra
+            #       fails to find a route, so I dumbly checked for 404?
+            if status != 404 && (routed = Thread.current[:datadog_http_routing].last)
+              # TODO: array == bad, this should be a Struct
+              last_script_name = routed[1]
+              last_route = routed[3]
+              # TODO: probably should get HTTP method from route resolvers as well
+
+              # TODO: I seem to have gathered that in some (all?) cases this
+              # should be nil when no route is found, which woudl cater for the
+              # note above
+              if last_route
+                # TODO: concatenate better? (according to spec I think it's fine /-wise)
+                composite_route = last_script_name + last_route
+                composite_path = last_script_name + last_route
+
+                request_span.set_tag(Tracing::Metadata::Ext::HTTP::TAG_ROUTE, composite_route)
+
+                # TODO: should it be composite_route?
+                request_span.resource = env['REQUEST_METHOD'] + ' ' + composite_path
```

Comment:
As is it doesn't work because of the various cases of resource setting in `ensure` below, but I believe this is what it should be on the Rack span, which handles HTTP.

I'm not sure about resource naming, it's all very confusing, especially when it ends up being `Controller#action` which to me makes no sense at all however I try to slice it.